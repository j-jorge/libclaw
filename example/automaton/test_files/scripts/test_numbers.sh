#!/bin/sh

./ex-automaton test_files/automaton_file/numbers.txt \
    238 +1289 -102 \
    .24 +.8921 -.0398 \
    239.23 +2323.19208 -092.23 \
    .24e12 +.8921e12098 -.0398e8372 \
    239.23e12 +2323.19208e1280 -092.23e12890 \
    .24E12 +.8921E12098 -.0398E8372 \
    239.23E12 +2323.19208E1280 -092.23E12890 \
    .24e+12 +.8921e+12098 -.0398e+8372 \
    239.23e+12 +2323.19208e+1280 -092.23e+12890 \
    .24E+12 +.8921E+12098 -.0398E+8372 \
    239.23E+12 +2323.19208E+1280 -092.23E+12890 \
    .24e-12 +.8921e-12098 -.0398e-8372 \
    239.23e-12 +2323.19208e-1280 -092.23e-12890 \
    .24E-12 +.8921E-12098 -.0398E-8372 \
    239.23E-12 +2323.19208E-1280 -092.23E-12890 \
    212. 3.232e 3.24E +42. -421. .32e+ .98E- +87.e 10fere
