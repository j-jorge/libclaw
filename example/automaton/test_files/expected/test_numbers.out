A = { +, -, ., 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, E, e }
I = { 0, 1 }
F = { 3, 4, 7 }
E = { 0, 1, 2, 3, 4, 5, 6, 7 }
T = {
0 -> { (1, +) (1, -) }
1 -> { (1, 0) (1, 1) (1, 2) (1, 3) (1, 4) (1, 5) (1, 6) (1, 7) (1, 8) (1, 9) (2, .) (4, 0) (4, 1) (4, 2) (4, 3) (4, 4) (4, 5) (4, 6) (4, 7) (4, 8) (4, 9) }
2 -> { (2, 0) (2, 1) (2, 2) (2, 3) (2, 4) (2, 5) (2, 6) (2, 7) (2, 8) (2, 9) (3, 0) (3, 1) (3, 2) (3, 3) (3, 4) (3, 5) (3, 6) (3, 7) (3, 8) (3, 9) }
3 -> { (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) }
4 -> { (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, E) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) (5, e) }
5 -> { (5, 0) (5, 1) (5, 2) (5, 3) (5, 4) (5, 5) (5, 6) (5, 7) (5, 8) (5, 9) (6, +) (6, -) (7, 0) (7, 1) (7, 2) (7, 3) (7, 4) (7, 5) (7, 6) (7, 7) (7, 8) (7, 9) }
6 -> { (6, 0) (6, 1) (6, 2) (6, 3) (6, 4) (6, 5) (6, 6) (6, 7) (6, 8) (6, 9) (7, 0) (7, 1) (7, 2) (7, 3) (7, 4) (7, 5) (7, 6) (7, 7) (7, 8) (7, 9) }
7 -> { }
}

238: valid
+1289: valid
-102: valid
.24: valid
+.8921: valid
-.0398: valid
239.23: valid
+2323.19208: valid
-092.23: valid
.24e12: valid
+.8921e12098: valid
-.0398e8372: valid
239.23e12: valid
+2323.19208e1280: valid
-092.23e12890: valid
.24E12: valid
+.8921E12098: valid
-.0398E8372: valid
239.23E12: valid
+2323.19208E1280: valid
-092.23E12890: valid
.24e+12: valid
+.8921e+12098: valid
-.0398e+8372: valid
239.23e+12: valid
+2323.19208e+1280: valid
-092.23e+12890: valid
.24E+12: valid
+.8921E+12098: valid
-.0398E+8372: valid
239.23E+12: valid
+2323.19208E+1280: valid
-092.23E+12890: valid
.24e-12: valid
+.8921e-12098: valid
-.0398e-8372: valid
239.23e-12: valid
+2323.19208e-1280: valid
-092.23e-12890: valid
.24E-12: valid
+.8921E-12098: valid
-.0398E-8372: valid
239.23E-12: valid
+2323.19208E-1280: valid
-092.23E-12890: valid
212.: not valid
3.232e: not valid
3.24E: not valid
+42.: not valid
-421.: not valid
.32e+: not valid
.98E-: not valid
+87.e: not valid
10fere: not valid
